common/djangoapps/third_party_auth/saml.py

```diff
@@ -41,10 +41,18 @@ def auth_url(self):
         if not self._config.enabled:
             log.error('SAML authentication is not enabled')
             raise Http404
+
         # TODO: remove this check once the fix is merged upstream:
         # https://github.com/omab/python-social-auth/pull/821
         if 'idp' not in self.strategy.request_data():
             raise AuthMissingParameter(self, 'idp')
+
+        from .models import SAMLProviderConfig
```

Comment:
Yes, I didn't look too hard into it, but the python path of this module at the time of execution doesn't contain this directory nor the common djangoapps when this module is loaded (I think this has to do with when python-social-auth djangoapp is initialized).
There's another preexisting location in this file which performs this import the same way for (presumably) the same reason.